src/store/mutations.ts

```diff
@@ -1,23 +1,103 @@
 import { CalChartState } from ".";
-import Show from "@/models/Show";
+import InitialShowState from "@/models/InitialShowState";
 import getters from "./getters";
 import BaseTool from "@/tools/BaseTool";
 import StuntSheet from "@/models/StuntSheet";
 import StuntSheetDot from "@/models/StuntSheetDot";
 import { MutationTree } from "vuex";
-import BaseCont from "@/models/continuity/BaseCont";
+import BaseCont, { CONT_IDS } from "@/models/continuity/BaseCont";
 import ContInPlace from "@/models/continuity/ContInPlace";
+import ContETFDynamic, {
+  ETF_DYNAMIC_TYPES,
+} from "@/models/continuity/ContETFDynamic";
 import DotAppearance from "@/models/DotAppearance";
+import { ContFactory } from "@/models/continuity/ContFactory";
+import { MARCH_TYPES } from "@/models/util/constants";
+import ContETFStatic from "@/models/continuity/ContETFStatic";
 
-const mutations: MutationTree<CalChartState> = {
-  // Show
-  setShow(state, show: Show): void {
-    state.show = show;
+export enum Mutations {
+  // Show mutations:
+  SET_SHOW = "Set new Show",
+  SET_SHOW_TITLE = "Set Show title",
+  ADD_STUNT_SHEET = "Add Stunt Sheet",
+  DELETE_STUNT_SHEET = "Delete Stunt Sheet",
+  // Field mutations:
+  SET_FRONT_HASH_OFFSET_Y = "Set front hash offset",
+  SET_BACK_HASH_OFFSET_Y = "Set back has offset",
+  SET_MIDDLE_OF_FIELD = "Set middle of field",
+  // Stuntsheet mutations:
+  ADD_DOTS = "Add Marcher",
+  REMOVE_DOTS = "Remove Marcher",
+  MOVE_DOTS = "Move Marcher",
+  SET_STUNT_SHEET_TITLE = "Set Stunt Sheet title",
+  SET_STUNT_SHEET_BEATS = "Set Stund Sheet beats",
+  ADD_DOT_TYPE = "Add Marcher type",
+  ADD_CONTINUITY = "Add Continuity",
+  // Continuity mutations:
+  UPDATE_DOT_TYPE_MARCH_STYLE = "Update Marcher Step Style",
+  UPDATE_DOT_TYPE_DURATION = "Update Marcher Duration",
+  UPDATE_DOT_TYPE_ETF_TYPE = "Update Marcher Eight To Five Flow",
+  UPDATE_DOT_TYPE_ETF_DIRECTION = "Update Marcher Flow Direction",
+  UPDATE_DOT_TYPE_IN_PLACE_DIRECTION = "Update Marcher Standing Direction",
```

Comment:
I think it would be easier if these 5 `UPDATE_DOT_TYPE`s were combined into a single `UPDATE_DOT_TYPE_CONTINUITY` that takes in a new updated continuity. That way for the more complicated continuities like follow the leader, step two, etc we wouldn't have to create new mutations for each different small change to the continuity.

Reply:
The challenge is what would be the parameter to such a function?  Currently what we do is we grab a continuity from the show state, modify it, and then commit it as a "continuity change".  But because we're modifying the state outside of `mutations`, and that's a no-no.  Mutations of the calchart state should go through the Vuex `mutations` path so that they can be undo-able.

We could enforce that by making is such that each mutation has to create a completely new clone of itself...  That may work.

Reply:
We can always refactor later to try to combine all the dot type mutations into one. For now, I'm ok with shipping this so we can get the undo/redo system going!

Reply:
> The challenge is what would be the parameter to such a function? Currently what we do is we grab a continuity from the show state, modify it, and then commit it as a "continuity change". But because we're modifying the state outside of `mutations`, and that's a no-no. Mutations of the calchart state should go through the Vuex `mutations` path so that they can be undo-able.
> 
> We could enforce that by making is such that each mutation has to create a completely new clone of itself... That may work.

That's what I was thinking too. Now that I think about it, that would make the dotTypeEditors much harder to figure out so I'm OK with what we've got here.